### ISSUE TITLE
[Regression] AutoSuggestBox, ComboBox, (probably others) no longer transform their popups correctly in WindowsAppSDK 1.4

### ISSUE THREAD

**Issue by garrettpauls** [microsoft/microsoft-ui-xaml#9118]
### Describe the bug

After updating from WindowsAppSDK 1.3 to 1.4, any AutoSuggestBox or ComboBox inside a container with a ScaleTransform applied to it do not scale their popups for suggestions or items correctly. The unscaled popups do, however, act as though the mouse cursor is hovering over the item where it would be if it _were_ scaled correctly, which is a big problem for us.

It would be tolerable if the scaled ComboBox showed an unscaled items popup if hovering the mouse over the items popup behaved by selecting the item that was actual under the mouse, however this is not the case.

### Steps to reproduce the bug

1. Add a ComboBox or AutoSuggestBox with items inside a container that has a ScaleTransform applied to it (it can be any level up the stack from the target control). Set this transform to have a ScaleX and/or ScaleY not equal to 1.0.
2. Run the application and observe the described behavior.

```xaml
<Window
	x:Class="PopupScaling.MainWindow"
	xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
	xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
	xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
	xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
	mc:Ignorable="d">

	<Grid>
		<Grid.ColumnDefinitions>
			<ColumnDefinition Width="*" />
			<ColumnDefinition Width="*" />
		</Grid.ColumnDefinitions>

		<StackPanel Grid.Column="0">
			<ComboBox Header="Normal ComboBox" ItemsSource="{x:Bind Values}" />

			<AutoSuggestBox Header="Normal AutoSuggestBox" ItemsSource="{x:Bind Values}" HorizontalAlignment="Left" />
		</StackPanel>

		<StackPanel Grid.Column="1">
			<StackPanel.RenderTransform>
				<ScaleTransform ScaleX="1.5" ScaleY="1.5" />
			</StackPanel.RenderTransform>

			<ComboBox Header="Scaled ComboBox" ItemsSource="{x:Bind Values}" />

			<AutoSuggestBox Header="Scaled AutoSuggestBox" ItemsSource="{x:Bind Values}" HorizontalAlignment="Left" />
		</StackPanel>
	</Grid>
</Window>
```

```cs
using System.Linq;
using Microsoft.UI.Xaml;

namespace PopupScaling;

public sealed partial class MainWindow : Window
{
	public MainWindow()
	{
		InitializeComponent();
	}

	public int[] Values { get; } = Enumerable.Range(1, 100).ToArray();
}
```

### Expected behavior

The expected behavior is for the popup to scale along with the control that creates it, and to show pointer over UI changes and respond to clicks based on where the user clicked relative to the visual control, not where the control thinks it is.

Even if the scaling can't be inherited by the popups, at the bear minimum the controls in the popup should respond to the mouse as though it had the same scaling as the popup, not the parent.

### Screenshots

In this gif, take note of where the mouse cursor is in relation to the highlighted list item in the ComboBox items list or the AutoSuggestBox suggestions list. The expected behavior is shown on the left, and the problem behavior is shown on the right.
![ScaledControlWithPopupBug](https://github.com/microsoft/microsoft-ui-xaml/assets/670089/2a25dcc2-6346-4bb8-a4fd-3d34d189b9f8)

### NuGet package version

WinUI 3 - Windows App SDK 1.4.2: 1.4.231008000

### Windows version

Windows 11 (21H2): Build 22000

### Additional context

Downgrading the app to use Microsoft.WindowsAppSDK version 1.3.230724000 fixes this issue, so it seems to be a regression in version 1.4.

**Comment by garrettpauls** [comment]
The culprit appears to be `ShouldConstrainToRootBounds="False"` being set on the Flyout and Popup controls used in the built-in control templates. I tried this out with a popup containing a list box, and it behaves exactly as expected when `ShouldConstrainToRootBounds` is unset, and breaks with the above problem as soon as `ShouldConstrainToRootBounds="False"` is set on it.

We could work around this by copying and modifying all the build in control templates to unset that value, but that feels very heavy-handed to fix something that should just work. At the very least the mouse pointer should behave as expected regardless of whether a popup is being scaled or not, or constrained to root bounds or not.

**Comment by gegao18** [comment]
Thanks for reporting this. We have a fix for this in the 1.6 timeframe.

**Comment by gegao18** [comment]
Update - the fix has been ported to 1.5.2.
